src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java

```diff
@@ -80,20 +94,49 @@ public void start() throws Exception {
             consumerMonitor.updateAndGet(monitor -> {
                 if (monitor == null) {
                     try {
-                        monitor = connectionManager.basicConsume(queue, this::handle, this::canceled);
+                        monitor = connectionManager.basicConsume(queue, this::handle, this::canceled, new MainMetrics(livenessMonitor, readinessMonitor));
                         LOGGER.info("Start listening exchangeName='{}', routing key='{}', queue name='{}'", exchangeName, routingKey, queue);
                     } catch (IOException e) {
                         throw new IllegalStateException("Can not start subscribe to queue = " + queue, e);
                     }
                 }
-
                 return monitor;
             });
-        } catch (Exception e) {
+        }  catch (IllegalStateException e) {
             throw new IOException("Can not start listening", e);
         }
     }
 
+    private void resubscribe() {
+        consumerMonitor.set(null);
+        ConnectionManager connectionManager = this.connectionManager.get();
+
+        int timeout = connectionManager.getMinConnectionRecoveryTimeout();
+
+        try {
+            ScheduledExecutorService executorService = executor.get();
+
+            if(executorService != null) {
+                executorService.submit(() -> {
+                    try {
+                        start();
+                        readinessMonitor.unregister(readinessMonitor.getId());
+                        livenessMonitor.unregister(livenessMonitor.getId());
+                    } catch (Exception e) {
+                        LOGGER.error(e.getMessage(), e);
+                        connectionManager.restoreChannel();
+
+                        livenessMonitor.disable();
+
+                        executorService.schedule(this::resubscribe, timeout, TimeUnit.MILLISECONDS);
+                    }
+                });
+            }
+        } catch (CancellationException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+    }
+
```

Comment:
I don't think it's a good idea. There're several reasons. First, it would not be correct if we move the logic of resubscribing from the class the whole point of which is subscribing. Second, we will have to register and unregister monitors in `ConnectionManager` specifically for this case.

Reply:
1. ConnectionManager already contains logic about reconnection, and it is not bad if we add resubscribe logic
2. We use too much method from connection manager. We can create method in it and change other methods modificators to private, and before call new method we can unregister monitors, and do other preparing things.

Reply:
I found one more issue. The first thing we do in this method is `consumerMonitor.set(null)`. In order to move `resubscribe()` to `ConnectionManager`, we will have to make `consumerMonitor` public, which is an `AtomicReference`. That definitely doesn't look good.

Reply:
We should not make consumerMonitor is public, because it change in our `start()` method. We can use `start()` like argument for new method in connection manager, like lambda.